openmmml/models/nequippotential.py

```diff
@@ -0,0 +1,198 @@
+"""
+nequippotential.py: Implements the NequIP potential function.
+
+This is part of the OpenMM molecular simulation toolkit originating from
+Simbios, the NIH National Center for Physics-Based Simulation of
+Biological Structures at Stanford, funded under the NIH Roadmap for
+Medical Research, grant U54 GM072970. See https://simtk.org.
+
+Portions copyright (c) 2021 Stanford University and the Authors.
+Authors: Peter Eastman
+Contributors: Stephen Farr
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS, CONTRIBUTORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+
+from openmmml.mlpotential import MLPotential, MLPotentialImpl, MLPotentialImplFactory
+import openmm
+from typing import Iterable, Optional, Union, Tuple
+from openmmml.models.utils import simple_nl
+
+class NequIPPotentialImplFactory(MLPotentialImplFactory):
+    """This is the factory that creates NequipPotentialImpl objects."""
+
+    def createImpl(self, name: str, model_path: str, distance_to_nm: float, energy_to_kJ_per_mol: float, atom_types: Optional[Iterable[int]]=None, **args) -> MLPotentialImpl:
+        return NequIPPotentialImpl(name, model_path, distance_to_nm, energy_to_kJ_per_mol, atom_types)
+
+class NequIPPotentialImpl(MLPotentialImpl):
+    """This is the MLPotentialImpl implementing the NequIP potential.
```

Comment:
We should clarify what this means, since there isn't really such a thing as "the NequIP potential".  NequIP is a code, not a potential.  Presumably this class can be used for any model implemented with that code, including Allegro models for example?

Reply:
The class can be used with any model generated by NequIP. I'm currently verifying if this also applies to Allegro, though I anticipate it should.